### PR TITLE
tests[contracts]: Improve Lib_MerkleTrie test coverage

### LINKED ISSUE #771: Improve test coverage in Lib_MerkleTrie
**Is your feature request related to a problem? Please describe.**
Our current test coverage for Lib_MerkleTrie is, IMO, insufficent. See [here](https://codecov.io/gh/ethereum-optimism/optimism/src/master/packages/contracts/contracts/optimistic-ethereum/libraries/trie/Lib_MerkleTrie.sol) for more detailed coverage info. I'd like to improve this.

**Describe the solution you'd like**
We just need to design some tests here to get coverage close to 100%. This is one of the most important libraries in our system.

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adding some tests to improve coverage for `Lib_MerkleTrie`, also now modifying the key/val sizes during tests to trigger some extra lines that deal with small nodes.

**Metadata**
- Fixes #771

### CODE REVIEW

File: packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_BytesUtils.sol [hunk 1]
```diff
@@ -93,7 +93,7 @@ library Lib_BytesUtils {
             bytes memory
         )
     {
-        if (_bytes.length - _start == 0) {
+        if (_start >= _bytes.length) {
```
Comment (gakonst) [inline packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_BytesUtils.sol:96]: Was this failing the tests before? Could we run in a bug in our currently deployed networks?

File: packages/contracts/contracts/optimistic-ethereum/libraries/trie/Lib_MerkleTrie.sol [hunk 2]
```diff
@@ -348,7 +348,32 @@ library Lib_MerkleTrie {
...
+                    _getNodeKey(lastNode),
+                    Lib_BytesUtils.slice(Lib_BytesUtils.toNibbles(_key), l)
+                ) == _getNodeKey(lastNode).length
+                && keyRemainder.length == 0
+            ) {
+                matchLeaf = true;
+            }
+        }
+
+        if (matchLeaf) {
```
Comment (gakonst) [inline packages/contracts/contracts/optimistic-ethereum/libraries/trie/Lib_MerkleTrie.sol:376]: Was there a bug here?

**Review by gakonst — COMMENTED** [review]
Tests look good - nice work on adding the official ones. Curious why the contract changes were required? Do we have bugs which can be hit in prod?

File: packages/contracts/contracts/optimistic-ethereum/libraries/trie/Lib_MerkleTrie.sol (hunk 2 above)
Comment (smartcontracts) [reply packages/contracts/contracts/optimistic-ethereum/libraries/trie/Lib_MerkleTrie.sol:376]: Yeah, we had a bug here that could potentially have caused a fraud proof to break

**Review by gakonst — APPROVED** [review]
This contract change COULD have affected mainnet/kovan fraud proofs. Since there have been no fraud proofs to date, this is good to go to master.

File: packages/contracts/test/data/json/libraries/trie/trietest.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "branch-value-update": {
+      "in": [
+        [ "abc", "123" ],
+        [ "abcd", "abcd" ],
+        [ "abc", "abc" ]
+      ],
+      "root": "0x7a320748f780ad9ad5b0837302075ce0eeba6c26e3d8562c67ccc0f1b273298a"
+    }
+  }
+}
```
Comment (maurelian) [inline packages/contracts/test/data/json/libraries/trie/trietest.json:26]: 👀

**Review by maurelian — CHANGES_REQUESTED** [review]
Just a couple nits that should be addressed first.

**Review by maurelian — APPROVED** [review]

**Pull request merged**